FEAT: Implement Skills marketplace for PyAEDT (#7586)
ansys/pyaedt: .github/plugin/pyaedt-cli/skills/SKILL.md

---
name: pyaedt-cli

> description: Use when interacting with ANSYS AEDT through the current PyAEDT CLI. Triggers on requests to start or inspect AEDT sessions, manage projects, create designs, run scripts, export screenshots or configs or use the PyAEDT command line from the workspace virtual environment.

Reviewer: ```suggestion
description: Use when interacting with ANSYS AEDT through the current PyAEDT CLI. Triggers on requests to start or inspect AEDT sessions, manage projects, create designs, run scripts, export screenshots or configurations, or use the PyAEDT command line from the workspace virtual environment.
```

---

# PyAEDT CLI skill

You have access to the `pyaedt` CLI to interact with ANSYS Electronics Desktop (AEDT). This skill teaches you how to use it effectively.

> ## Pre-Flight Check (Run First)

Reviewer: In documentation we try to follow google rules and only have a single capital letter in the titles (to be applied in all sections)

Before any CLI command, verify `pyaedt` is installed:

```bash
pyaedt --json version
```

> If the command fails, stop and tell the user to install it first:

Reviewer: ```suggestion
If the command fails, stop and instruct the user to install it first using the `all` installation target to enable full capabilities.
```

JSON output returns: `{"status": "ok", "data": {...}}` or `{"status": "error", "error": "message"}`

Always parse the `status` field before using `data`. Never rely on human-readable output.

Exceptions:

- `session attach` launches an interactive console, so human mode is acceptable.
- `doc` commands open browser pages and are not usually consumed as structured automation output.

## Current CLI Shape

Commands are organized as:

```bash
pyaedt [--json] <group> <command> [options]
```

Available groups in the current CLI:

- `session`
- `project`
- `script`
- `export`
- `test-config`
- `panels`
- `doc`

Top-level commands:

- `version`
- `aedt-versions`

Do not invent other command groups or commands that do not exist in the current CLI.
If a required operation is not supported by the CLI, report that to the user instead of trying to work around it.

## Port-Based Model

The current CLI is port-based, not session-file based.

- `session start` launches a new AEDT instance and returns its gRPC port.
- `session list` discovers running AEDT instances.
- Most operational commands require `--port` explicitly.

> - There is no persisted `session connect` state to reuse later.

Reviewer: This comment is a bit akward given that `session attach` can work without the port and as some selection mechanism.

It opens an iPython console that is already connected to the target AEDT session — `desktop` is pre-injected, so PyAEDT commands can be issued one by one without writing a full script.

```bash
# Attach to a running instance (port optional — prompts if omitted)
pyaedt session attach --port 50051

# Inside the iPython console — the session is already live:
# >>> import ansys.aedt.core as aedt
# >>> hfss = aedt.Hfss(new_desktop=False, port=50051)
# >>> box = hfss.modeler.create_box([0, 0, 0], [10, 10, 10], name="MyBox")
# >>> hfss.modeler["MyBox"].color = (255, 0, 0)
```

#### Sending Multi-Line Code to the iPython Console

When the agent sends code to the iPython console that spans **more than one line** (loops, conditionals, function definitions, etc.), pasting it directly causes an `IndentationError` or `SyntaxError` because the console treats each line as a separate input.

**Always wrap multi-line blocks in `exec()`**, encoding all newlines as `\n` inside a single string:

```python
# Fails — the console breaks on the loop body
for i in range(3):
    print(i)

# Correct — send as a single exec() call

> exec("for i in range(3):\n    print(i)")

Reviewer: I'm not sure about the context of use of that SKILL.md file. It can't execute code without user's approbation, right ? Cause calling `exec(...)` would effectively allow execution of arbitrary Python code in the iPython session.

Author: Code is executed using an IPython session previously launched. Every time the CLI executes a command, as running code in a terminal, it asks for user permission. So running this code is final user's choice

```

Rules:

- Use `exec()` for **any** block that contains indented code (loops, `if`, `with`, function or class definitions).
- Single-statement lines (imports, assignments, function calls) do **not** need `exec()`.
- Escape all inner double-quotes as `\"` if the outer delimiter is `"`, or use concatenated string literals as shown above.
- Keep `exec()` calls as short as possible; if the block grows beyond ~10 lines prefer saving a temp script and using `run` instead.

> **When to prefer `session attach` over `run`:**

Reviewer: ```suggestion
**When to prefer `session attach` command over `run` command:**
```

# Step 3: run the script on the chosen port
pyaedt --json run "<temp-dir>/generated_script.py" --port 50051
```

Rules:

- Always determine the target port first.
- If multiple instances are running, present the full list and ask which port to use.
- Do not tell the user to run the generated script manually if the workflow requires automatic execution.
- Generated scripts intended for an existing AEDT instance should use `new_desktop=False`.

### Workflow 7: Export Design Data

```bash
# If the instance has exactly one project and one design, selectors can be omitted
pyaedt --json export screenshot --port 50051 --path "<temp-dir>/preview.jpg"
pyaedt --json export config --port 50051 --output "<temp-dir>/config.json" --overwrite

# If ambiguous, specify the missing selector(s)
pyaedt --json export screenshot --port 50051 --project "PatchAntenna" --design "HFSSDesign1" --path "<temp-dir>/preview.jpg"
```

## Error Handling

Always parse the `status` field in JSON output before using `data`:

```json
{"status": "ok", "data": {...}}
{"status": "error", "error": "message"}
```

If `status` is `"error"`, report the `error` message to the user and stop. Do not attempt to use `data`.

> ## Guidance Rules For This Skill

Reviewer: Feels like the items are duplicate of previous statements, should we keep it ?

Author: This is just a summary of all the previous indications. I would probably keep it as it doesn't add a lot of extra context. But it's on us to test them and decide if those lines are not necessary. 
My experience is that providing a short workflow summary after defining the skill indications works solidly
